src/easylink/step.py

```diff
@@ -1300,63 +1325,23 @@ def set_configuration_state(
             The configuration for any implementations to be combined.
         input_data_config
             The input data configuration for the entire pipeline.
-
-        Notes
-        -----
-        We update the :class:`easylink.graph_components.StepGraph` and ``SlotMappings``
-        in :meth:`validate_step` (as opposed to in :meth:`set_configuration_state`
-        as is done in :class:`TemplatedStep`) because :meth:`validate_step` is called
-        prior to :meth:`set_configuration_state`, but the validations itself actually
-        requires the updated ``StepGraph`` and ``SlotMappings``.
         """
+        chosen_config = self.choices[step_config["type"]]
```

Comment:
I dislike the name of this variable, since it sounds like it comes from the user config. Perhaps just `choice`?